### PR TITLE
Cassandra collection tests refactor

### PR DESCRIPTION
**Pull request by conorbros**
This PR refactors all Cassandra collection tests to have multiple elements in each collection for testing.

Removed boolean because sets require unique values and it's adequately covered by the other types. Not worth handling the edge case.

### CODE REVIEW

File: shotover-proxy/tests/cassandra_int_tests/collections.rs [hunk 1]
```diff
@@ -350,42 +546,130 @@ mod map {
...
+        })
+        .await;
+        insert_table(connection, "map_list", |elements| {
+            let lists = format!("[{}]", elements.join(","));
+            let mut map = String::from("{");
+
+            for i in 0..2 {
+                map.push_str(&format!("{}: {}, ", i, lists));
+            }
+            map.push_str(&format!("{}: {}}}", 2, lists));
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/collections.rs:579]: would this do the same thing?
```rust
            for i in 0..3 {
                map.push_str(&format!("{}: {}, ", i, lists));
            }
            map.push_str("}");
```

Comment (conorbros) [reply shotover-proxy/tests/cassandra_int_tests/collections.rs:579]: That would leave a trailing comma, assumed invalid syntax but I can double check.

Comment (rukai) [reply shotover-proxy/tests/cassandra_int_tests/collections.rs:579]: oh its that trailing comma, isnt it?

**Comment by rukai** [comment]
Now we dont test boolean types at all, that seems unfortunate.
It should be straightforward to change `insert` and friends to work on an arbitrarily long `elements` instead of hardcoding to 3.
Then we can just have boolean contain 2 values (true and false).

File: shotover-proxy/tests/cassandra_int_tests/collections.rs (hunk 1 above)
Comment (conorbros) [reply shotover-proxy/tests/cassandra_int_tests/collections.rs:579]: How does it work for maps?

Comment (rukai) [reply shotover-proxy/tests/cassandra_int_tests/collections.rs:579]: I'll give it a go and report back.

Comment (rukai) [reply shotover-proxy/tests/cassandra_int_tests/collections.rs:579]: Here it is for map_list:

```rust
        insert_table(connection, "map_list", |elements| {
            let lists = format!("[{}]", elements.join(","));
            format!("{{{}}}", (0..3).map(|i| format!("{i}: {lists}")).join(","))
        })
        .await;
```

I also edited the example for set_set that I gave, there is hardcoded number of 3 outer sets that has no relation to the len of elements.

File: shotover-proxy/tests/cassandra_int_tests/collections.rs [hunk 2]
```diff
@@ -350,42 +504,125 @@ mod map {
...
+                elements
+                    .iter()
+                    .enumerate()
+                    .map(|(i, element)| { format!("{i}: {element}") })
+                    .join(",")
+            );
+
+            format!("{{{}}}", (0..3).map(|i| format!("{i}: {maps}")).join(","))
+
+            // let mut maps = String::from("{");
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/collections.rs:540]: dead code

**Comment by rukai** [comment]
I pushed a commit to fix the boolean issue.
Should be good to go now.

**Review by rukai — APPROVED** [review]

File: shotover-proxy/tests/cassandra_int_tests/collections.rs [hunk 3]
```diff
@@ -57,57 +57,148 @@ fn get_type_str(col_type: ColType) -> &'static str {
...
+            ),
+            ResultValue::TimeUuid(
+                uuid::Uuid::parse_str("84196262-53de-11ec-bf63-0242ac130002").unwrap(),
+            ),
+            ResultValue::TimeUuid(
+                uuid::Uuid::parse_str("84196262-53de-11ec-bf63-0242ac130003").unwrap(),
+            ),
+        ],
+        ColType::Inet => vec![
+            ResultValue::Inet("127.0.0.1".parse().unwrap()),
```
Comment (cjrolo) [inline shotover-proxy/tests/cassandra_int_tests/collections.rs:178]: Inet is only IPv4?

Comment (rukai) [reply shotover-proxy/tests/cassandra_int_tests/collections.rs:178]: https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/cql_data_types_c.html 
image

IPV4 or IPV6

**Review by cjrolo — APPROVED** [review]
